lib/vault/encrypted_model.rb

```diff
@@ -267,12 +252,16 @@ def __vault_persist_attribute!(attribute, options)
 
         # Only persist changed attributes to minimize requests - this helps
         # minimize the number of requests to Vault.
-        if !changed.include?("#{attribute}")
-          return
+        if ActiveRecord.gem_version >= Gem::Version.new("5.2")
```

Comment:
I thinks it would be a good idea to use Rails [previous_changes](https://github.com/rails/rails/blob/6-0-stable/activemodel/lib/active_model/dirty.rb#L227) here.

`previous_changes` exists from Rails 3.0 and it works on the latest Rails 6(master) also.

`previous_changes_include?` is dropped in Rails 5.2.3

Something like this:
```ruby
previous_changes.include?(attribute)
```

Reply:
Method https://github.com/rails/rails/blob/6-0-stable/activemodel/lib/active_model/dirty.rb#L227 based on `changes`, but `changes` didn't work for attributes like this.

Reply:
The behave differently, `changes` returns hash of changes before save, `previous_changes` after it's saved.

```ruby
person = Person.first
person.name = "New name"
person.changes  #=> {"name"=>["Old name", "New name"]}
person.previous_changes #=> {}

person.save

person.changes #=> {}
person.previous_changes  #=> {"name"=>["Old name", "New name"],...}


```

Reply:
I've just pull this PR and test it against rails master. With `previous_changes_include?` it doesn't work, with `previous_changes` works as expected

Reply:
This is not my pull request, I have other one: 
https://github.com/hashicorp/vault-rails/pull/76/files

I try show previous_changes here https://github.com/hashicorp/vault-rails/pull/76/files#diff-ba19de511d4dbf96b22d51bf1ecbf448R268 and it was empty when changed_attributes present.

But may be it specific only for my PR.

Reply:
@dixpac thanks for the suggestion, and you're right, if I make that change and get the latest Rails 6 RC into the Appraisals file, the test suite is green.

@justincampbell did you want the Rails 6 fix added to this PR? Or would you prefer to merge what's here in first, and then I can create a smaller one for just Rails 6?

Reply:
Hello @pat and @justincampbell we are using this branch since September and would like to know if there is a todolist of outstanding items for the merge, now we already need support for Rails 6 as well.

We can work on this pull request if needed

Reply:
Hello!  I'm interested in using this in production as well, but also would need this merged to master.  Can I help at all?

Reply:
Hello @pat Is there any work left to proceed with the merge ?

Reply:
@mod if you're finding it works for you, then I think it's ready? I'm not an owner of the repo, I can't merge things.

And also: the organisation I worked for when creating this PR no longer exists, and so I'm no longer using this gem – so your experience with it is going to be far more valid and recent than mine.